From 3144d3dd87ca071f56196c2507ee061800222ab4 Mon Sep 17 00:00:00 2001
From: Guo Li <guo.li@mediatek.com>
Date: Wed, 13 Mar 2019 18:02:33 +0800
Subject: [PATCH 16/19] ActivityThread: Prevent surface hang up when screen on
 and screen off

Set screen lock as "None", then open a app like settings,
Then do screen off and screen on by press powser key.
Then do screen off and screen on again, then the app can't be slided.

That's because the visibility of the activity didn't change,
and ViewRootImpl will not clear SOFT_INPUT_IS_FORWARD_NAVIGATION.
So when resume activity, AMS will not call updateViewLayout().

Bug: 112574560
Change-Id: If273d0971e22756f36046e7e44d157f9f24ef11d
Signed-off-by: rohan <purohit.rohan@gmail.com>
---
 core/java/android/app/ActivityThread.java | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/core/java/android/app/ActivityThread.java b/core/java/android/app/ActivityThread.java
index 24408f2225..212f5b2355 100644
--- a/core/java/android/app/ActivityThread.java
+++ b/core/java/android/app/ActivityThread.java
@@ -4920,13 +4920,14 @@ public final class ActivityThread extends ClientTransactionHandler
                 l.softInputMode = (l.softInputMode
                         & (~WindowManager.LayoutParams.SOFT_INPUT_IS_FORWARD_NAVIGATION))
                         | forwardBit;
-                if (r.activity.mVisibleFromClient) {
-                    ViewManager wm = a.getWindowManager();
-                    View decor = r.window.getDecorView();
-                    wm.updateViewLayout(decor, l);
-                }
             }
 
+            if (r.activity.mVisibleFromClient) {
+                ViewManager wm = a.getWindowManager();
+                View decor = r.window.getDecorView();
+                wm.updateViewLayout(decor, l);
+             }
+
             r.activity.mVisibleFromServer = true;
             mNumVisibleActivities++;
             if (r.activity.mVisibleFromClient) {
-- 
2.34.1

